src/main/webapp/script.js

```diff
@@ -215,6 +221,35 @@ function createCalendar() {
     });
 }
 
+function createGrocery() {
+    const groceryForm = document.getElementById('groceryFormID');
+
+    // creating FormData to get the values of the form
+    const formData = new FormData(groceryForm);
+    var queryString = "";
```

Comment:
You can declare the queryString inside the for block, using `let` or `const` instead of var

Or just not use a helper variable at all, and just do

array,push(expresion for creating the next queryString bit)

Reply:
solved